### ISSUE TITLE
Option to decouple certain things (like entering a camera) from host_timescale

### ISSUE THREAD

**Issue by TiberiumFusion** [Facepunch/garrysmod-requests#2038]
Experience the problem firsthand:
- Create a camera
- `host_timescale 0.01`
- Press the camera's key to enter the camera's view. It probably didn't work, right?
- Press and hold the camera's key for at least 1.666 seconds (1/60 fps * 1/0.01). Now the view switches to the camera.

This is a point of contention, since timescale is intended to affect the speed at which the world simulates. Buttons, emitters, lights, lamps, and cameras are all part of the world and are thus subject to timescale. Agreement with this can be argued one of two ways, depending on which conceptual idea of the world is being subscribed to for any given situation.

**A)** If a camera exists in the world, and the world is 100x slower than real time, then I should indeed have to hold down the camera key on my keyboard for 100x the usual duration in order to see this change occur in the game. This is "physically" correct if we treat my keyboard inputs as events occurring within the gmod world. My player had to hold his virtual camera key for a full 16.666 ms to get into his camera, so I need to hold my real keyboard key for 1.666 seconds for this to translate into 16.666 ms of ingame world time.

Or,

**B)** The camera I placed in the world may exist in the world, but me, the guy seated at his desk & keyboard, is not physically bound to that world. I do not have to wait 100x longer than realtime in order to push the key on my keyboard; ergo, if I, the guy seated at his desk & keyboard, want to look through a placed camera, the logic of the slowed down world should not leak out into the real world and require me to hold down my keyboard key for 1.666 seconds. This is "physically" correct if we treat my keyboard inputs as events occurring outside the gmod world, which is what they truly are.

The choice between the two is probably best represented by a simple question: is the gamemode competitive? If we're playing darkrp at 0.01 timescale (for torture, I suppose), it would be weird if I could gather a great deal of information by rapidly switching through 10 cameras before my player has even finishing taking a step. But, if I'm setting up a scene to take a screenshot or do some other kind of offline render, I don't want to have to hold down my camera, light, lamp, etc keys for 100x longer than I need to, especially if that shot needed involves switching cameras during the slow-mo sequence.

**tl;dr It would be great if there was an option, possibly a cvar, that decouples the timescale from whatever ugly chunk of code is applying it to player inputs. Treat player inputs as occurring outside of the world, like they actually do, instead of as if they occurred inside the world.**

**Comment by pparadoxx** [comment]
> Experience the problem firsthand:
> 
> * Create a camera
> * `host_timescale 0.01`
> * Press the camera's key to enter the camera's view. It probably didn't work, right?
> * Press and hold the camera's key for at least 1.666 seconds (1/60 fps * 1/0.01). Now the view switches to the camera.
> 
> This is a point of contention, since timescale is intended to affect the speed at which the world simulates. Buttons, emitters, lights, lamps, and cameras are all part of the world and are thus subject to timescale. Agreement with this can be argued one of two ways, depending on which conceptual idea of the world is being subscribed to for any given situation.
> 
> **A)** If a camera exists in the world, and the world is 100x slower than real time, then I should indeed have to hold down the camera key on my keyboard for 100x the usual duration in order to see this change occur in the game. This is "physically" correct if we treat my keyboard inputs as events occurring within the gmod world. My player had to hold his virtual camera key for a full 16.666 ms to get into his camera, so I need to hold my real keyboard key for 1.666 seconds for this to translate into 16.666 ms of ingame world time.
> 
> Or,
> 
> **B)** The camera I placed in the gmod world may exist in the gmod world, but me, the guy seated at his desk & keyboard, is not physically bound to that world. I do not have to wait 100x longer than realtime in order to push the key on my keyboard; ergo, if I, the guy seated at his desk & keyboard, want to look through a placed camera, the logic of the slowed down world should not leak out into the real world and require me to hold down my keyboard key for 1.666 seconds. This is "physically" correct if we treat my keyboard inputs as events occurring outside the gmod world, which is what they truly are.
> 
> The choice between the two is probably best represented by a simple question: is the gamemode competitive? If we're playing darkrp at 0.01 timescale (for torture, I suppose), it would be weird if I could gather a great deal of information by rapidly switching through 10 cameras before my player has even finishing taking a step or aimed his gun. But, if I'm setting up a scene to take a screenshot or do some other kind of offline render, I don't want to have to hold down my camera, light, lamp, etc keys for 100x longer than I need to, especially if that shot needed involves switching cameras during the slow-mo sequence.
> 
> **tl;dr It would be great if there was an option, possibly a cvar, that decouples the timescale from whatever ugly chunk of code is applying it to player inputs. Instead of treating player inputs as if they occur inside the gmod world, treat them as if they occur outside of the gmod world, like they actually do.**

Uh, it worked fine for me; maybe an addon you have? Also why the long winded explanation.

**Comment by TiberiumFusion** [comment]
You must be new here; this is standard fare for gmod politics. Also why the dismissive minimalist response?

**Comment by garryspins** [comment]
Hes very new round these parts, this would be a good addition and i can see how itd come in handy in more places than just with cameras
